components/tools/OmeroM/src/annotations/unlinkAnnotation.m

```diff
@@ -0,0 +1,166 @@
+function unlinkAnnotation(session,parentType,parentID,varargin)
+% unlinkAnnotation allows you to unlink all or specified
+% MapAnnotation (Key-Value Pairs in GUI) from parent in OMERO server.
+%
+% SYNTAX
+% unlinkAnnotation(session,parentType,parentID)
+% unlinkAnnotation(session,parentType,parentID,annt)
+%
+%
+% BEFORE USE
+% 
+%   client = loadOmero('demo.openmicroscopy.org', 4064)
+%   username = 'xxxxxxx'
+%   password = 'xxxxxxx'
+%   session = client.createSession(username, password)
+%
+%
+% AFTER USE
+%
+%   clear
+%   unloadOmero
+%
+% REQUIREMENTS
+%
+% OMERO.matlab toolbox
+% https://docs.openmicroscopy.org/omero/5.4.6/developers/Matlab.html
+% 
+%
+% INPUT ARGUMENTS
+% session     omero.api.ServiceFactoryPrxHelper object
+%
+%               client = loadOmero('demo.openmicroscopy.org', 4064)
+%               session = client.createSession('knakamura', 'omeroomero')
+%
+% parentType  'project' | 'dataset' | 'image' | 'screen' | 'plate' | 
+%             'plateacquisition' | 'roi'
+%
+% parentID    vector of non-negative integers
+%             parentID can be found via OMERO GUI
+%
+%
+% annt        omero.model.Annotation object or its array | [] (default)
+%
+%             (Optional) omero.model.Annotation object (supports
+%             omero.model.MapAnnotationI, omero.model.TagAnnotationI,
+%             omero.model.CommentAnnotationI, and
+%             omero.model.FileAnnotationI at the moment) to be unlinked
+%             from parent If empty or not specified, all linked
+%             MapAnnotation objects will be unlinked.
+%            
+%             For example, to obtain MapAnnotationI object(s) from an image:
+%
+%               annt = getObjectAnnotations(session, 'map', 'image', parentID);
+%
+%             or
+%
+%               annt = getImageTagAnnotations(session, ids)
+% 
+%             annt can be empty, scalar or a vector.
+%
+%             Also you can use following functions or similar.
+%
+%               annt = getAnnotations(session,ids,'tags')
+%               annt = getTagAnnotations(session,tagID)
+%
+%
+%
+% Written by Kouichi C. Nakamura Ph.D.
+% MRC Brain Network Dynamics Unit
+% University of Oxford
+% kouichi.c.nakamura@gmail.com
+% 09-Jun-2018 03:27:05
+%
+% See also
+% linkAnnotation
+
+objects = getObjectTypes();
+
+p = inputParser;
+p.addRequired('session',@(x) isjava(x));
+p.addRequired('parentType',@(x) ischar(x) && ismember(x, {objects.name}));
+p.addRequired('parentID',@(x) isnumeric(x) && isvector(x));
+
+p.addOptional('annt',[],@(x) isempty(x) || all(isa(x, 'omero.model.Annotation')));
+p.parse(session,parentType,parentID,varargin{:});
+
+annt = p.Results.annt;
+
+x = annt;
+assert(isa(x,'omero.model.MapAnnotationI') || isa(x,'omero.model.TagAnnotationI') ...
```

Comment:
For sure you can get rid of line 90. I personally cannot test all other types of Annotation objects so I narrowed it down just to be conservative.

Reply:
Thank you for the time and effort.

Should I change the code accordingly now?

Reply:
Yes please feel free to push the changes to this branch and we can retest. If we agree the example addition covers everything, let me know if you want to cherry-pick sbesson/openmicroscopy@268fb4c so that the function is tested by our daily Continuous Integration jobs. I can also open it as a separate PR.

Reply:
I didn't know how to cherry pick to that branch so I added the changes manually. Hope they're in good shape.